### ISSUE TITLE
Got reported OpenCV exception

### ISSUE THREAD

**Issue by sleeplessai** [AlexeyAB/darknet#5522]
Hi, @AlexeyAB .

I built darknet by using **vcpkg** and successfully installed as shown below: 
```
Computing installation plan...
The following packages are already installed:
    darknet[cuda,cudnn,opencv-cuda,opencv-base,core]:x64-linux
Package darknet:x64-linux is already installed

Total elapsed time: 5.031 us

The package darknet:x64-linux provides CMake targets:

    find_package(Darknet CONFIG REQUIRED)
    target_link_libraries(main PRIVATE Darknet::dark)
```
But during test `./darknet detector test cfg/coco.data cfg/yolov4.cfg weights/yolov4.weights data/horses.jpg`, I got: 
```
 CUDA-version: 10010 (10020), cuDNN: 7.6.0, GPU count: 1  
 OpenCV version: 4.1.1
......
Total BFLOPS 128.459 
avg_outputs = 1068395 
 Allocate additional workspace_size = 52.43 MB 
Loading weights from weights/yolov4.weights...
 seen 64, trained: 32032 K-images (500 Kilo-batches_64) 
Done! Loaded 162 layers from weights-file 
data/horses.jpg: Predicted in 44.084000 milli-seconds.
horse: 77%
horse: 98%
horse: 90%
horse: 87%
horse: 100%
OpenCV exception: show_image_cv 
OpenCV exception: wait_key_cv 
OpenCV exception: destroy_all_windows_cv 
```
though the `prediction.jpg` drew bounding boxes and classification tags correctly.

And during training `./darknet detector train  cfg/coco.data yolov4a.cfg backup/yolov4a_last.weights `, I got: 
```
v3 (iou loss, Normalizer: (iou: 0.07, cls: 1.00) Region 139 Avg (IOU: -nan, GIOU: -nan), Class: -nan, Obj: -nan, No Obj: 0.000248, .5R: -nan, .75R: -nan, count: 0, class_loss = 0.005503, iou_loss = 0.000000, total_loss = 0.005503 
v3 (iou loss, Normalizer: (iou: 0.07, cls: 1.00) Region 150 Avg (IOU: 0.502149, GIOU: 0.431751), Class: 0.325852, Obj: 0.001273, No Obj: 0.000872, .5R: 0.600000, .75R: 0.200000, count: 5, class_loss = 70.260963, iou_loss = 3.380981, total_loss = 73.641945 
v3 (iou loss, Normalizer: (iou: 0.07, cls: 1.00) Region 161 Avg (IOU: 0.255511, GIOU: 0.207969), Class: 0.340541, Obj: 0.003271, No Obj: 0.004223, .5R: 0.052632, .75R: 0.000000, count: 19, class_loss = 197.708389, iou_loss = 0.982010, total_loss = 198.690399 
OpenCV exception: wait_key_cv
```
How to solve these OpenCV exceptions if not changing the way to build ?

**Comment by GustavoStahl** [comment]
I'm having the same issue here, i recompiled OpenCV from source and reinstalled the darknet repo, but it didn't change 
``` 
CUDA-version: 10020 (10020), cuDNN: 7.6.5, GPU count: 1  
OpenCV version: 4.3.0
0 : compute_capability = 610, cudnn_half = 0, GPU: GeForce GTX 1050 
net.optimized_memory = 0 
..................................
Total BFLOPS 65.879 
avg_outputs = 532444 
Allocate additional workspace_size = 52.43 MB 
Loading weights from yolov3.weights...
seen 64, trained: 32013 K-images (500 Kilo-batches_64) 
Done! Loaded 107 layers from weights-file 
data/dog.jpg: Predicted in 65.362000 milli-seconds.
bicycle: 99%
dog: 100%
truck: 94%
OpenCV exception: show_image_cv 
OpenCV exception: wait_key_cv 
OpenCV exception: destroy_all_windows_cv 
```

**Comment by sleeplessai** [comment]
@GustavoStahl 
I do the same `vcpkg install` on my Windows 10 and everything works well.
I got exceptions again after I recompiled `darknet[cuda,cudnn,opencv-base]` on Linux.

**Comment by GustavoStahl** [comment]
@sleeplessai 
I found a way to make OpenCV work, don't know if it will help you. I built from source OpenCV for the 4th time and set `NVCC=/usr/local/cuda-10.2/bin/nvcc` instead of `NVCC=nvcc` in Darknet MakeFile. Still can't train any model, it just closes the loss chart after training starts without any errors, if you have any suggestions

**Comment by cenit** [comment]
One big warning: the code in vcpkg is still lagging behind and might not be perfectly yolov4 compatible. Unfortunately a bug in cmake has also been exposed (we are very bleeding edge in its usage) and this is making the update very long and tedious.
I suggest you to try to append a `--head` to the Darknet install command line on vcpkg
`./vcpkg install darknet[features] --head` in order to have the most recent code
Again, sorry for the trouble, but I am pushing for fixes upstream daily and we will get to the end!

**Comment by sleeplessai** [comment]
@cenit @GustavoStahl 
Sorry, I'm late to reply. I figured it out on my Ubuntu 20.04 LTS, just then.
The reason why OpenCV exception occurs may be Linux cannot locate the OpenCV installed in vcpkg. 

I remain there is a pkg-config  `.pc`  configuration file, when compile OpenCV from sources, in that Linux is able to find built libopencv library and its headers. I searched for .pc file of OpenCV in vcpkg directory but not found, so I deprecated vcpkg. Instead, I did `sudo apt install libopencv-dev` to get OpenCV 4.2.0 because I think the system may certainly find the packages from itself.

I switched to compile darknet from sources and I am going to `git pull` everyday. I keep the default settings and compile it by doing `>sh build.sh`. 

Finally, there is no OpenCV exception any more. 
// I can see my cute dog in data/dog.jpg at last.

**Comment by cenit** [comment]
vcpkg uses cmake config files to find OpenCV, not pkg-config files to find it.
I am working on OpenCV 4.3 right now, it will be published on vcpkg soon.
If you want to try again with it, when it will be ready, I'd be glad. Happy that you found a solution in the meantime, anyway

**Comment by sleeplessai** [comment]
@cenit 
Thanks for working for that. I will keep it stable on Linux and try latest vcpkg on Windows.
By the way, I should focus on training networks.
